std/exception.d

```diff
@@ -1613,3 +1614,462 @@ version(unittest) package
     static assert({ cast(void)dg(); return true; }());
     cast(void)dg();
 }
+
+/** This $(D enum) is used to select the methods of the range to handle by the
+  $(D handle) range wrapper. The values of the $(D enum) can be OR'd to
+  selected multiple primitives to be handled.
+ */
+enum RangePrimitive
+{
+    front    = 0b00_0000_0001,
+    back     = 0b00_0000_0010,
+    popFront = 0b00_0000_0100,
+    popBack  = 0b00_0000_1000,
+    empty    = 0b00_0001_0000,
+    save     = 0b00_0010_0000,
+    length   = 0b00_0100_0000,
+    opDollar = 0b00_1000_0000,
+    opIndex  = 0b01_0000_0000,
+    opSlice  = 0b10_0000_0000,
+}
+
+/** This range handles exceptions originating in ranges.
+
+To use the range the code has to specify what methods of the range should be
+handled. This is done by use of the $(LREF $(D RangePrimitive)) enum. Multiple
+methods of a range can be handled at once, by using the or operator two
+combine multiple $(D RangePrimitive). If more than one method should be
+handled by one user supplied handler, all methods must have the same sigature.
+
+The $(D handler) $(D alias) must takes a $(D Throwable) as first of type $(D
+R) as first argument a range of type $(D IRange) as second argument. The
+return type of the $(D alias) must be equal to the return type of the
+primitive handled.
+
+Params:
+    E = The type of Throwable to handle.
+    functionsToHandle = The $(D RangePrimitive) selecting which primitives to
+        handle.
+    handler = The callable that is called when a handled primitive throw an
+    $(D Throwable) of type $(D E). The handler must have the same return type
+    as the handled primitive and must accepect parameters of the $(D E, ref
+    IRange).
+    input = The range to handle.
+
+Returns: A wrapper $(D struct) that preserves the $(D range) interface of the
+passed $(D IRange input).
+*/
+auto handle(E : Throwable, RangePrimitive functionsToHandle, alias handler, IRange)(IRange input)
+if (isInputRange!IRange)
+{
+    static struct Handler(E : Throwable, alias handler, IRange)
+    {
+        IRange range;
+
+        static if (isForwardRange!IRange)
+        {
+            static if (functionsToHandle & RangePrimitive.save)
+            {
+                @property typeof(this) save()
+                {
+                    try
+                    {
+                        return typeof(this)(range.save());
```

Comment:
Since `.save` is `@property`, it should not be invoked with parentheses, i.e., should be `range.save` instead of `range.save()`.